phases/19-capstone-projects/75-end-to-end-eval-runner/code/main.py

```
@@ -0,0 +1,382 @@
+        print(f"  {d['model_a']} vs {d['model_b']}: diff={d['diff_mean']:+.3f}  ci=[{d['ci_lo']:+.3f},{d['ci_hi']:+.3f}]  significant={sig}")
+
+    if not report.leaderboard:
+        print("ERROR: empty leaderboard")
+        return 1
+    top = report.leaderboard[0]["model_id"]
+    bot = report.leaderboard[-1]["model_id"]
+    if top != "rule_based":
+        print(f"ERROR: expected rule_based at top, got {top}")
+        return 2
+    if bot == "rule_based":
+        print(f"ERROR: rule_based should not be at the bottom")
```

Ruff (0.15.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)